kits/java/src/main/java/com/luxai/lux/objectmapper/Mapper.java

```diff
@@ -0,0 +1,74 @@
+package com.luxai.lux.objectmapper;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.luxai.Agent;
+import com.luxai.lux.State;
+
+import java.util.Map;
+
+public class Mapper {
+
+    public static String getJson(Object object) throws JsonProcessingException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        return objectMapper.writeValueAsString(object);
+    }
+
+    public static void updateState(Agent agent, String json) throws JsonProcessingException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        // check first step
+        if (agent.obs == null) {
+            json = json
+                    .replace("rubble", "rubble_init")
+                    .replace("lichen_strains", "strains_init")
+                    .replace("lichen", "lichen_init");
+
+            State state = objectMapper.readValue(json, State.class);
+            agent.obs = state.obs;
+            agent.step = state.step;
+            agent.remainingOverageTime = state.remainingOverageTime;
+            agent.player = state.player;
+            agent.env_cfg = state.info.env_cfg;
+        }
+        else {
+            State state = objectMapper.readValue(json, State.class);
+            agent.step = state.step;
+            agent.remainingOverageTime = state.remainingOverageTime;
+            agent.obs.teams = state.obs.teams;
+            agent.obs.real_env_steps = state.obs.real_env_steps;
+            agent.obs.units.clear(); agent.obs.units = state.obs.units;

```

Comment:
I don't see a need for clearing it before assignment. Java should garbage collect the old stuff anyway.

Reply:
fix